src/components/NamespaceDropdown.tsx

```diff
@@ -20,32 +20,52 @@ export class NamespaceDropdown extends React.PureComponent<NamespaceListType, {}
     this.props.refresh();
   }
 
-  handleSelectNamespace = (namespace: string) => {
-    this.props.onSelect({ name: namespace });
+  onChange = (a: any) => {
+    this.props.onSelect({ name: a.target.value });
   };
 
+  namespaceButtonText() {
+    if (this.props.activeNamespaces.length === 0) {
+      return 'Select a namespace';
+    } else if (this.props.activeNamespaces.length === 1) {
+      return this.props.activeNamespaces[0].name;
+    } else {
+      return `${this.props.activeNamespaces.length} selected`;
+    }
+  }
+
   handleToggle = (isOpen: boolean) => isOpen && this.props.refresh();
 
   render() {
-    const disabled = this.props.disabled ? true : false;
```

Comment:
Did you mean to get rid of disabled?

Reply:
I think i'm going to get rid of it after the rebase

Reply:
Why get rid of it?  Is it because with the cancelablePromises we feel it's OK to change the setting during an update?

Reply:
Exactly, currently it will update when you click a namespace, disabling would block the user if he wants to use multiple namespaces. 

Even if we changed to a "wait before actually changing the state" it would add additional waiting time if the user takes long to decide what namespace or if the auto-refresh triggers while we have this open.

I think the way to go would be to [cancel](https://github.com/axios/axios#cancellation) the current loading and do a new one with the update params, that way we ensure that there is no in-between data.

I think we could even show the GraphLoading screen if we detect we are loading a different namespace set to avoid any confusion on what graph we are looking at.

Reply:
Sounds good to me.